server/routes/choropleth.py

```diff
@@ -35,85 +35,109 @@
 
     # Download statistical variable, names, and geojson for subgeos.
     stat_var_by_geo = dc.get_stats(geos_contained_in_place,
-                                   stat_var_requested)
+                                   requested_stat_var)
     names_by_geo = dc.get_property_values(geos_contained_in_place,
                                           "name")
     geojson_by_geo = dc.get_property_values(geos_contained_in_place,
                                             "geoJsonCoordinates")
 
     # Download population data if per capita.
-    population_by_geo = {} if not per_capita else \
-        dc.get_stats(geos_contained_in_place, "Count_Person")
+    population_by_geo = ({} if not per_capita
+        else dc.get_stats(geos_contained_in_place, measurement_denominator))
 
-    # Process into combined json object.
+    # Process into a combined json object.
     features, values = [], []
     for geo_id, json_text in geojson_by_geo.items():
         # Valid response needs at least geometry and a name.
-        if len(json_text) > 0 and geo_id in names_by_geo:
+        if json_text and geo_id in names_by_geo:
             geo_feature = {
                     "type": "Feature",
                     "geometry": {
                         "type": "MultiPolygon",
                     },
                     "id": geo_id,
                     "properties": {
+                        # Choose the first name when multiple are present.
                         "name": names_by_geo[geo_id][0],
-                        "hasSublevel": (geolevel_below != "City"),
+                        "hasSublevel": 
+                            (display_level in LEVEL_MAP),
                         "geoId": geo_id,
                     }
                 }
             # Load, simplify, and add geoJSON coordinates.
+            # First returned value is chosen always. 
             geojson = json.loads(json_text[0])
-            geo_feature['geometry']['coordinates'] = \
+            geo_feature['geometry']['coordinates'] = (
                 coerce_geojson_to_righthand_rule(
                                     geojson['coordinates'],
-                                    geojson['type'])
+                                    geojson['type']))
             # Process Statistical Observation if valid.
-            if geo_id in stat_var_by_geo and \
-                    (not per_capita or geo_id in population_by_geo):
+            if (geo_id in stat_var_by_geo and
+                    'data' in stat_var_by_geo[geo_id] and
+                    (not per_capita or
+                    (geo_id in population_by_geo and
+                    'data' in population_by_geo[geo_id]))):
                 # Grab the latest available data.
-                stat_obs = next(iter(stat_var_by_geo[geo_id]['data'].values()))
+                stat_obs = next(iter(
+                    reversed(stat_var_by_geo[geo_id]['data'].values())))
```

Comment:
Are the [] operations guaranteed to succeed?  We do not want the program to crash.  I think we are given the conditional.

Reply:
Yes at this point